### PR TITLE
miscellaneous fixes

### PR DESCRIPTION
**Pull request by jonsterling**

### CODE REVIEW

File: hedberg-wip.red [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import bool
+
+;; WIP
+
+data void where
```
Comment (favonia) [inline hedberg-wip.red]: This is so Haskell!

**Comment by jonsterling** [comment]
By the way, the real challenge with improving subtyping is to carry over those improvements to the unifier-based subtyping algorithm, which is shit.

I want to have a system with just one subtyping algorithm, and it should emit constraints that are then passed to a very lightweight unifier.

**Comment by favonia** [comment]
Is there an example to show off the new algorithm?

**Comment by jonsterling** [comment]
@favonia Sadly any example I can think of would require unfucking the unifier-based-subtyping algorithm...

But the intention is to have better behavior wrt eta expansion, so that one doesn't need to manually eta expand elements of extension type.

**Pull request merged**